### PR TITLE
Makes ORCA CASSCF parsing work for optimizations and open shells

### PR DESCRIPTION
**Pull request by jevandezande**
When performing geometry optimizations and open-shell computations, the printing in ORCA changes slightly. This PR fixes them and removes the previous error warnings.

### CODE REVIEW

File: cclib/parser/orcaparser.py [hunk 1]
```diff
@@ -771,9 +771,12 @@ def splitter(line):
                 self.etsecs.append(sec)
                 line = next(inputfile)
 
-
         # Parse the various absorption spectra for TDDFT and ROCIS
         if 'ABSORPTION SPECTRUM' in line or 'ELECTRIC DIPOLE' in line:
+            # CASSCF has an anomalous printing of ABSORPTION SPECTRUM
+            if line[:-1] == 'ABSORPTION SPECTRUM':
+                return
```
Comment (berquist) [inline cclib/parser/orcaparser.py]: Why isn't this `break`?

Comment (jevandezande) [reply cclib/parser/orcaparser.py]: It isn't inside a loop. The function extract is called inside a loop over the lines of the inputfile (line 311 of logpfileparser.py)

File: cclib/parser/orcaparser.py [hunk 2]
```diff
@@ -1223,37 +1226,37 @@ def energy_intensity(line):
...
+            self.skip_lines(inputfile, ['b', 'd'])
+            if next(inputfile).strip() == 'ORBITAL ENERGIES':
+                self.skip_lines(inputfile, ['d', 'b', 'NO'])
+                orbitals = []
                 vals = next(inputfile).split()
-
-            self.skip_line(inputfile, 'b')
+                while vals:
+                    occ, eh, ev = map(float, vals[1:4])
+                    # The irrep will only be printed if using symmetry
```
Comment (langner) [inline cclib/parser/orcaparser.py]: Please end sentences in comments with periods.

File: cclib/parser/orcaparser.py [hunk 3]
```diff
@@ -771,9 +771,12 @@ def splitter(line):
                 self.etsecs.append(sec)
                 line = next(inputfile)
 
-
         # Parse the various absorption spectra for TDDFT and ROCIS
         if 'ABSORPTION SPECTRUM' in line or 'ELECTRIC DIPOLE' in line:
+            # CASSCF has an anomalous printing of ABSORPTION SPECTRUM
```
Comment (langner) [inline cclib/parser/orcaparser.py]: Does anomalous mean we can't parse it?

**Comment by jevandezande** [comment]
I'm ashamed of my grammar! I'm headed out on vacation, I'll fix it when I
get back in a week.

The ABSORPTION SPECTRUM line is printed but no spectrum is printed (see
line 1127 of H2-_CASSCF_opt.out in PR 81 in cclib data). I will ask the
maintainer of the CASSCF module why it is printed.

Karol M. Langner <notifications@github.com> schrieb am Do., 31. Mai 2018,
19:08:

**Comment by langner** [comment]
> I'm ashamed of my grammar! I'm headed out on vacation, I'll fix it when I
> get back in a week.

Don't be ashamed. I don't doubt your grammar skills, I think it is simply a matter of habit.

**Comment by langner** [comment]
> I've added a pull request to cclib-data repo wherein H2- is optimized.

FTR, this is in cclib/cclib-data#81

File: cclib/parser/orcaparser.py (hunk 1 above)
Comment (langner) [reply cclib/parser/orcaparser.py]: Eric's point if that break allows the parser to potentially trigger off the same line... so I also think `break` is better.

**Comment by langner** [comment]
@jevandezande Did you plan on working on this?

**Comment by jevandezande** [comment]
If @berquist can remove his change request (satisfied by [cclib/cclib-data#81](https://github.com/cclib/cclib-data/pull/81)), this should be able to be merged in. Is there anything else that needs to be added?

**Comment by berquist** [comment]
I don't think so; anything else will have to wait until we start defining actual CASSCF attributes.

**Comment by berquist** [comment]
> Did you want to add a regression test in this PR?

I didn't see this, but confirmed the regression output didn't even parse properly without the PR. Do you still want one?

**Pull request merged**